# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1301589266 25200
#      Thu Mar 31 09:34:26 2011 -0700
# Node ID 422a7a7e9b6ec788f95750077287f721b89ea0df
# Parent  4040a95a586d0361ae16b9e914b61145f436bc37
cor.m: fix operation with only single input (bug #32961)

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,11 +1,12 @@
 2011-03-31  Rik  <octave@nomad.inbox5.com>
 
-	* plot/private/__quiver__.m: Revamp autoscaling algorithm (bug #32836).
+	* statistics/base/cor.m: Fix operation with only single input
+	(bug #32961)
 
 2011-03-29  John W. Eaton  <jwe@octave.org>
 
 	* special-matrix/wilkinson.m: Update test for 'wilkinson(1)' to reflect
 	changes to diag().
 
 2011-03-29  Michael Creel  <michael.creel@uab.es>
 
diff --git a/scripts/statistics/base/cor.m b/scripts/statistics/base/cor.m
--- a/scripts/statistics/base/cor.m
+++ b/scripts/statistics/base/cor.m
@@ -20,18 +20,27 @@
 ## @deftypefn  {Function File} {} cor (@var{x})
 ## @deftypefnx {Function File} {} cor (@var{x}, @var{y})
 ## Compute matrix of correlation coefficients.
 ##
 ## This is an alias for @code{corrcoef}.
 ## @seealso{corrcoef}
 ## @end deftypefn
 
-function retval = cor (x, y = [])
+function retval = cor (x, y = x)
 
   if (nargin < 1 || nargin > 2)
     print_usage ();
   endif
 
   retval = corrcoef (x, y);
 
 endfunction
 
+%!test
+%! x = rand (10, 2);
+%! assert (isequal (cor (x), corrcoef (x)));
+%! assert (cor (x(:,1), x(:,2)) == corrcoef (x(:,1), x(:,2)));
+
+%% Test input validation
+%!error corrcoef ();
+%!error corrcoef (1, 2, 3);
+
